src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -705,6 +707,8 @@ private function registerDebugConfiguration(array $config, ContainerBuilder $con
 
         if ($debug) {
             $container->setParameter('debug.container.dump', '%kernel.cache_dir%/%kernel.container_class%.xml');
+            $container->register(RobotsTagListener::class)
+                ->addTag('kernel.event_listener', ['event' => KernelEvents::RESPONSE, 'priority' => -255]);
```

Comment:
This PR is for a "Dev" environment, but it's only added when in debug mode, there's some inconsistency here. Should it be registered when debug is on or when in dev mode? I'd expect the latter, so debug would be incorrect here.

Reply:
I think debug is a safer choice. If for some fancy reason your app is hosted in test mode, might be a better idea not to index it.

Reply:
I think debug shouldn't influence this setting _at all_. 

However, it would make sense to allow this in production for systems that don't need indexing, like a customer portal. This is neither in dev nor debug. What about a configuration option that you can define per environment instead?

Reply:
Now linked to a `disallow_search_engine_index` configuration key in the framework bundle